app/models/aws_engine.rb

```diff
@@ -0,0 +1,275 @@
+class AWSEngine
+  class Config
+    include ActiveSupport::Configurable
+    config_accessor :bucket_name
+    config_accessor :image_bucket_name
+    config_accessor :use_s3_static_hosting
+    config_accessor :region
+    config_accessor :cdn_base_url
+    config_accessor :sqs_url
+    config_accessor :aws_access_id
+    config_accessor :aws_secret_key
+  end
+
+  def self.configure(&block)
+    yield config
+
+    if self.configured?
+      Aws.config.update({
+        region: @config.region,
+        credentials: Aws::Credentials.new(@config.aws_access_id, @config.aws_secret_key),
+      },)
+    end
+    @client = Aws::S3::Client.new(region: @config.region)
+  end
+
+  def self.config
+    @config ||= Config.new
+  end
+
+  def self.configured?
+    return false unless defined? @config.aws_access_id
+    return false unless defined? @config.aws_secret_key
+    return false if @config.aws_access_id.blank?
+    return false if @config.aws_secret_key.blank?
+    true
+  end
+
+  def self.resource_endpoint
+    unless @config.cdn_base_url.blank?
+      url = @config.cdn_base_url
+    else
+      if @config.use_s3_static_hosting == '1'
```

Comment:
__[rubocop]__
* Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.